### ISSUE TITLE
borg hangs on extract of a single (small) file

### ISSUE THREAD

**Issue by LastStarDust** [borgbackup/borg#4943]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes. There are some issues opened for very large archives and very large files. But this is not my case.

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE report

## System information. For client/server mode post info for both machines.

Client
OS: Ubuntu 18.04 64bit
Borg: 1.1.10

Server
OS: Scientific Linux 6.10 64bit
Borg: 1.1.10

#### Hardware / network configuration, and filesystems used.

Client:
Lenovo Carbon X1 gen6
FS: ext4 on NVMe SSD

Server:
Cloud computing cluster
FS: IBM GPFS (on tape?)

They are connected by a double SSH link
Client -> Intermediate machine -> Server

#### How much data is handled by borg?
About 100 archives in the same repository.
A single archive is about 15GB.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
```
/usr/bin/borg extract --progress --log-json kekcc:/hsm/nu/wagasci/data/beam::physics_run_2019-11-06_21-33-11_5 home/wagasci-ana/Data/beam/physics_run_2019-11-06_21-33-11_5/physics_run_2019-11-06_21-33-11_5.log
```

## Describe the problem you're observing.

When I try to extract a single XML file (a few kB in size) from SOME of the archives in the same repository, borg hangs indefinitely (at least more than half an hour), while for some other archives, under the very same network conditions, the extraction is completed in a few seconds. 

All the archives were created with the very same command and have about the same file content and file size (about 10 big files (GB size) and about 100 little files (kB size)).

The output XML file is actually created but it is empty.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

The problem is always the same for all the offending archives. I tried to extract the files inside the server and the problem was the same (so maybe not a connection issue).

#### Include any warning/errors/backtraces from the system logs

There seem to be no network or CPU usage due to borg when it hangs. The following log includes me pressing CTRL-C to put borg out of its misery.

```
/usr/bin/borg extract --progress --log-json kekcc:/hsm/nu/wagasci/data/beam::physics_run_2019-11-06_21-33-11_5 home/wagasci-ana/Data/beam/physics_run_2019-11-06_21-33-11_5/physics_run_2019-11-06_21-33-11_5.log
{"message": "Calculating size", "current": 0, "total": 0, "info": null, "operation": 1, "msgid": "extract", "type": "progress_percent", "finished": false, "time": 1581390351.8807251}
^C{"type": "log_message", "time": 1581390712.3617225, "message": "Connection closed by remote host", "levelname": "ERROR", "name": "borg.archiver", "msgid": "ConnectionClosed"}
```

**Comment by ThomasWaldmann** [comment]
I don't think borg recognizes that: `kekcc:`

We either use `ssh:` or `file:` or a simple fs path.

**Comment by LastStarDust** [comment]
`kekcc` is a SSH hostname defined in my `.ssh/config`. I assure you that it is not that the issue. The same command works for other archives.

**Comment by ThomasWaldmann** [comment]
The syntax for ssh is either `ssh://...` or `user@host:...`, see the docs for details.

Not sure it gets recognized correctly if there is no `user@` part.

**Comment by ThomasWaldmann** [comment]
And if you can reproduce the issue without involving ssh, network and tape, please do so and remove all that stuff from this ticket.

**Comment by LastStarDust** [comment]
I have done as you asked. I tried to extract the file in the very same machine (server) where the repository resides. All the paths in the issue description are local. For some of the archives the extraction is rather quick (seconds), for some others it hangs (or is slower than 30 minutes).

**Comment by ThomasWaldmann** [comment]
Thanks! Does it also happen if you copy the repo to an ext4 / zfs or other known-good normal fs?

**Comment by LastStarDust** [comment]
To test that it might take some time. But I will let you know.

**Comment by ThomasWaldmann** [comment]
Any news?

**Comment by LastStarDust** [comment]
Sorry I kind of forgot about this issue report.
So well ... I was not able to reproduce the problem in a reliable way.
It might have been due to a transient/temporary connection issue or server issue.
Today I tested it again and was able to download 100 of the affected files without major problems.
Moreover, I have implemented in my application a caching mechanism to reduce to the minimum the queries to the remote borg repository, thus speeding up the application and minimizing these connection issues.
I am closing the bug report. Sorry for taking so much time.
Thank you
